### PR TITLE
Fix transform of DeterministicSingleTimeSeries with one window

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/time_series_parameters.jl [hunk 1]
```diff
@@ -71,7 +71,11 @@ function TimeSeriesParameters(
 )
     last_timestamp = initial_timestamp + resolution * len
     last_initial_time = last_timestamp - resolution * horizon
-    count = length(range(initial_timestamp, step = interval, stop = last_initial_time))
+    if interval == Dates.Second(0)
+        count = 1
+    else
+        count = length(range(initial_timestamp, step = interval, stop = last_initial_time))
```
Comment (jd-lara) [inline src/time_series_parameters.jl:77]: do we need to allocate the range to calculate this count? isn't easier to do `(last - initial)/interval`

File: src/utils/utils.jl [hunk 2]
```diff
@@ -367,7 +367,7 @@ function get_initial_times(
     if count == 0
         return []
     elseif count == 1
-        @assert interval == Dates.Second(0)
+        @assert interval == Dates.Second(0) "interval=$interval"
```
Comment (jd-lara) [inline src/utils/utils.jl:370]: I have never seen this syntax of @assert and then string. Does this work?

**Review by jd-lara — CHANGES_REQUESTED** [review]

File: src/utils/utils.jl (hunk 2 above)
Comment (daniel-thom) [reply src/utils/utils.jl:370]: yes

File: src/time_series_parameters.jl (hunk 1 above)
Comment (daniel-thom) [reply src/time_series_parameters.jl]: There was actually an off-by-one error here. Fixed.

File: src/utils/utils.jl (hunk 2 above)
Comment (jd-lara) [reply src/utils/utils.jl:370]: That's cool. It will make it far easier to understand

Comment (daniel-thom) [reply src/utils/utils.jl:370]: What we should do is make a macro that does this for you automatically.

File: src/time_series_parameters.jl (hunk 1 above)
Comment (jd-lara) [reply src/time_series_parameters.jl]: How about the comment on the count calculation?

Comment (daniel-thom) [reply src/time_series_parameters.jl]: GitHub seems confused on this comment. I did remove the range call, so I'm not sure what you mean. Can you review the code again? I did just add an explanatory comment, but I'm not sure you meant that.

**Review by jd-lara — APPROVED** [review]
LGTM

**Pull request merged**